### PR TITLE
Restrict creating organisations by user

### PR DESCRIPTION
**Pull request by Schwobaland**
Allows to restrict, wether an user is allowed to create organisations. See #177.

### CODE REVIEW

File: conf/locale/locale_de-DE.ini [hunk 1]
```diff
@@ -815,6 +815,7 @@ team_permission_desc=Welche Berechtigungsstufe soll das Team haben?
 
 form.name_reserved=Organisationsname '%s' ist bereits vergeben.
 form.name_pattern_not_allowed=Organisationsnamen der Form '%s' sind nicht erlaubt.
+form.create_org_not_allowed=Dieser Benutzer darf keine Organisation erstellen.
```
Comment (strk) [inline conf/locale/locale_de-DE.ini]: I think you're supposed to only modify the english localization, whereas other languages are delegated to an external service. Please @lunny, @bkcsoft, @tboerger  correct me if I'm wrong

File: models/migrations/v15.go [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+type UserV15 struct {
+	AllowCreateOrganization bool
+}
+
+func (*UserV15) TableName() string {
+	return "user"
+}
+
+func createAllowCreateOrganizationColumn(x *xorm.Engine) error {
+	return x.Sync2(new(UserV15))
```
Comment (strk) [inline models/migrations/v15.go]: What's the default value set for existing users upon migration ?

File: models/migrations/migrations.go [hunk 3]
```diff
@@ -18,7 +18,7 @@ import (
 	"github.com/Unknwon/com"
 	"github.com/go-xorm/xorm"
 	gouuid "github.com/satori/go.uuid"
-	"gopkg.in/ini.v1"
+	ini "gopkg.in/ini.v1"
```
Comment (strk) [inline models/migrations/migrations.go]: If this isn't needed for your patch please avoid the change

**Review by strk — CHANGES_REQUESTED** [review]
Please don't include bindata.go modifications in the PR

File: conf/locale/locale_de-DE.ini (hunk 1 above)
Comment (Schwobaland) [reply conf/locale/locale_de-DE.ini]: I didn't find the translation service for gitea. There is only Crowdin for [gogs](https://crowdin.com/project/gogs) and an empty project for [gitea](https://crowdin.com/project/gitea).

Comment (tboerger) [reply conf/locale/locale_de-DE.ini]: We want to use an external service again but it's not enabled yet

File: models/migrations/migrations.go (hunk 3 above)
Comment (tboerger) [reply models/migrations/migrations.go]: As long as it's done in a separate commit it's OK. It's an goimports correction.

**Comment by tboerger** [comment]
Please never update vendored libs, it needs to be integrated upstream first and than updated via govendor

**Comment by Schwobaland** [comment]
Thanks for your reviews. I reverted the changes you suggested to remove. The field AllowCreateOrganization is set to true in migration script.
Please review again.

**Comment by tboerger** [comment]
It already looks pretty fine, but you want to set a default value to true, otherwise registered users are not allowed to create organizations

**Comment by Schwobaland** [comment]
added default value both in  `CreateUser`-function and in xorm-annotation

**Comment by strk** [comment]
It would be useful if the site administrator was given the option to decide which permissions to give by default to new users.

**Comment by tboerger** [comment]
This PR needs to be delayed until the branch of 1.0 anyway.

**Comment by bkcsoft** [comment]
LGTM. While I do agree with @strk, that can be implemented at a later point :)

File: models/migrations/migrations.go [hunk 4]
```diff
@@ -72,8 +72,10 @@ var migrations = []Migration{
 
 	// v13 -> v14:v0.9.87
 	NewMigration("set comment updated with created", setCommentUpdatedWithCreated),
-
+	// v14
 	NewMigration("create user column diff view style", createUserColumnDiffViewStyle),
+	// v15
+	NewMigration("create user colum allow create organization", createAllowCreateOrganizationColumn),
```
Comment (ethantkoenig) [inline models/migrations/migrations.go]: `"... column ..."`?

File: models/user.go [hunk 5]
```diff
@@ -185,6 +186,11 @@ func (u *User) CanCreateRepo() bool {
 	return u.NumRepos < u.MaxRepoCreation
 }
 
+// CanCreateOrg returns true if user can create organisation.
```
Comment (ethantkoenig) [inline models/user.go]: `CanCreateOrganization`?

**Comment by tboerger** [comment]
Please resolve the conflicts

File: models/migrations/migrations.go (hunk 4 above)
Comment (Schwobaland) [reply models/migrations/migrations.go]: fixed typo

File: models/user.go (hunk 5 above)
Comment (Schwobaland) [reply models/user.go]: fixed typo

**Comment by Schwobaland** [comment]
conflicts resolved

**Comment by lunny** [comment]
LGTM

**Pull request merged**